### DISCUSSION TITLE
Calling volumetric function with cameras from ImplicitRenderer

### DISCUSSION THREAD

**Discussion by YanivHollander** [facebookresearch/pytorch3d#1080] [Q&A] [answered]
The method `forward` in class `ImplicitRenderer` calls on line 168 the `volumetric_function` model with cameras as an input argument. The `forward` of `volumetric_function` (in class `NeuralRadianceFieldm`, e.g.) does not have cameras in its API, nor does it need it. Rays are already well defined at this point, and the volumetric function can return opacity and features along the rays, without the need for the camera objects that were used to sample the rays. 

Please advise, and if needed I will create an issue on this.

**Comment by bottler** [comment, marked as answer]
ImplicitRenderer is generic - designed to work with many different raysamplers, volumetric functions, and raymarchers. In general, implementations of these all need to ignore extra keyword arguments. You could imagine a use case where the volumetric function needs the cameras, so we provide it. The NeRF project is just an example of the use of ImplicitRenderer which doesn't use it. This isn't a bug.
